### PR TITLE
ECER-1165: account setup page to collect ECE number

### PR DESCRIPTION
**Pull request by farzadnadiri**
## Title
ECER-1165: Account setup page to collect ECE number

## Related Jira Issue(s)

- ECER-1165

## Checklist
- [x] I have tested these changes locally.
- [ ] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

## Screenshots (if applicable)
image

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue [hunk 1]
```diff
@@ -1,38 +1,78 @@
...
+          <ECEHeader title="Contact Information" />
+          <p class="small mt-2">We'll use this to contact you about your account and updates about your application or certificate.</p>
+          <v-text-field
+            v-model="email"
+            class="mt-5 max-width-500"
+            label="Email"
+            variant="outlined"
+            color="primary"
+            type="email"
+            :rules="emailRules"
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:30]: We can use the centralized formRules file. 

I think this page was done before we had this in place. 

So for example, this emailRules could be written as 

import * as Rules from "formrules"

[Rules.required('You must enter email whatever"), Rules.email()]

you can take a look how we implemented this here https://github.com/bcgov/ECC-ECER/blob/b4148409b408509130f04fb44fa1d828dfe33323/src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue#L26

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue [hunk 2]
```diff
@@ -1,38 +1,78 @@
...
+          <v-text-field
+            v-if="eceCertificateStatus === true"
+            v-model="eceRegistrationNumber"
+            label="Your ECE Registration Number"
+            variant="outlined"
+            color="primary"
+            class="max-width-300"
+            :rules="eceRegistrationRules"
+            @keypress="isNumber($event)"
+          />
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:60]: AC: 
The input should be a text input, not a numeric one. The number may be too large for a numeric field and has ux issues with accidentally changing the number in a numeric field.

I think we should copy what we have done for other fields that accept numbers. Where we restrict their input. So they won't need that numbers rule. 

Ex. https://github.com/bcgov/ECC-ECER/blob/b4148409b408509130f04fb44fa1d828dfe33323/src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue#L41

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/oidc.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export const useOidcStore = defineStore("oidc", {
     userManager: new UserManager(useConfigStore().kcOidcConfiguration),
   }),
   actions: {
-    async oidcUserInfo(): Promise<Components.Schemas.UserInfo> {
+    async oidcUserInfo(): Promise<any> {
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/oidc.ts:18]: just wanted to confirm that we want this to be any?

Comment (farzadnadiri) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/oidc.ts:18]: I might change the user info object in ECER-1166 and roll back this to user info based on server schema, but for this story we need to show the user address and that's why i changed it to any

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue (hunk 2 above)
Comment (farzadnadiri) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:60]: I've already added this: @keypress="isNumber($event)" see line 59, I just get rid of this one:

(v: string) => {
          if (this.eceCertificateStatus === true) {
            return /^\d+$/.test(v) || "Your ECE registration number must be a number"; // Must be a number if "Yes" is selected
          }
          return true; // No validation if "No" selected
        },

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue [hunk 4]
```diff
@@ -106,38 +136,12 @@ export default defineComponent({
...
-        },
-      ];
-    },
-    radioRules() {
-      return [
-        (v: boolean | null) => v !== null || "Choose an option", // Radio button required
-      ];
-    },
-  },
+  computed: {},
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:143]: we can remove computed if there's nothing in it.

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue [hunk 5]
```diff
@@ -1,78 +1,108 @@
...
+          </v-row>
+        </v-form>
+      </v-col>
+
+      <!-- Buttons -->
+      <v-col cols="12">
+        <v-row>
+          <v-btn rounded="lg" color="primary" class="mr-2" @click="submit">Save and continue</v-btn>
+          <v-btn rounded="lg" variant="outlined" @click="logout">Cancel</v-btn>
+        </v-row>
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:103]: It's good practice to always use v-row with v-col together. Noticed v-row without v-col for the buttons.

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue [hunk 6]
```diff
@@ -1,78 +1,108 @@
...
+            <v-col cols="12" sm="6">
+              <v-text-field
+                v-model="phoneNumber"
+                label="Phone number"
+                variant="outlined"
+                color="primary"
+                :rules="[Rules.required(), Rules.phoneNumber()]"
+                @keypress="isNumber($event)"
+              ></v-text-field>
+            </v-col>
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:54]: Do the email and phone number cols settings work as expected? 

From how I read it, in a large screen, email and phone number would appear side by side like this 

image

then between the md size screen and small size screen email would span the entire row and phone number would take half like this 

image

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue (hunk 5 above)
Comment (farzadnadiri) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:103]: Replaced with div!

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue (hunk 4 above)
Comment (farzadnadiri) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:143]: Removed!

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue (hunk 6 above)
Comment (farzadnadiri) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:54]: somehow it was not happening before, but I put those into different v-rows, solved the problem. but I believe the first solution 'max-width' was a better way to handle this as we did not need columns and rows here. it also was supporting responsive.

Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:60]: @farzadnadiri sorry noticed I put the wrong photo for the example. Updated it

Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/NewUser.vue:60]: yeah putting it into separate v-rows fixes the problem. 

It gets funky with cols when we mix and match the numbers expecting it to show up in separate rows.

**Review by SoLetsDev — APPROVED** [review]

**Pull request merged**